packages/oats-runtime/src/make.ts

```diff
@@ -580,9 +667,20 @@ function fromUnionReflection(types: Type[]): Maker<any, any> {
     // note that we need to check the undiscriminated values also as those *might* match also
     discriminatedMakers.set(key, makeOneOf(...[...untaggedMaker, fromReflection(t)]));
   }
+  // we choose the first entry from discriminators as all the options *must* have the same property name and
+  // thus the same network property name.
+  const networkDiscriminatorKey =
+    networkMapFromDiscriminatedType(discriminator.key, [...discriminator.map.values()][0]) ??
+    discriminator.key;
   return (value: any, opts?: MakeOptions) => {
-    if (value && typeof value === 'object' && !Array.isArray(value) && discriminator.key in value) {
-      const discriminatedType = discriminatedMakers.get(value[discriminator.key]);
+    const key = getInputPropName({
```

Comment:
the value may have been made already (in case we are in an allOf) so need to check those in case the discriminator key has been converted already.